### PR TITLE
NumPy course changes, May 2018

### PR DESCRIPTION
**Pull request by dkillick**
Changes to the NumPy course based on the recent run of the course.

Ping @corinnebosley @kaedonkers

### CODE REVIEW

File: course_content/notebooks/numpy_intro.ipynb [hunk 1]
```diff
@@ -267,9 +255,9 @@
     "\n",
     "##### Part 1\n",
     "\n",
-    "Use the code cells below to explore features of these arrays.\n",
+    "Use the code cells below to explore features of this array.\n",
     "\n",
-    "* What is the `type` of each of the arrays created above? (Hint: you can find the type of an object in Python using `type(<object>)`, where `<object>` is replaced with the name of the variable containing the object.)\n",
+    "* What is the `type` of each of the array created above? (Hint: you can find the type of an object in Python using `type(<object>)`, where `<object>` is replaced with the name of the variable containing the object.)\n",
```
Comment (pp-mo) [inline course_content/notebooks/numpy_intro.ipynb]: Created a grammar problem : ? should it be "of each array" ?

Comment (dkillick) [reply course_content/notebooks/numpy_intro.ipynb]: Looks like it! Thanks @pp-mo

File: course_content/notebooks/numpy_intro.ipynb [hunk 2]
```diff
@@ -113,6 +99,8 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
+    "#### Exercise\n",
```
Comment (kaedonkers) [inline course_content/notebooks/numpy_intro.ipynb]: @LukeC92 has made some changes to the numbering of the exercises in PR #136.
I would suggest changing this to "Exercise 1" and merging this PR before #136.

File: course_content/notebooks/numpy_intro.ipynb [hunk 3]
```diff
@@ -303,9 +291,7 @@
     "For each of these properties:\n",
     "\n",
     "* Use information from the array documentation to find out more about each array property.\n",
-    "* Apply each property to the array `arr` defined above. Can you explain the results you get in each case?\n",
-    "\n",
-    "Be ready to feed your results back to the group!"
```
Comment (kaedonkers) [inline course_content/notebooks/numpy_intro.ipynb]: :open_mouth: :cake: :fork_and_knife: ?

File: course_content/notebooks/numpy_intro.ipynb [hunk 4]
```diff
@@ -1515,9 +1499,7 @@
     "\n",
     "### Concepts:\n",
     "\n",
-    "  * [masked arrays](https://docs.scipy.org/doc/numpy/reference/maskedarray.generic.html) and missing data:\n",
```
Comment (kaedonkers) [inline course_content/notebooks/numpy_intro.ipynb]: Why were these removed?

File: course_content/notebooks/numpy_intro.ipynb [hunk 5]
```diff
@@ -1527,7 +1509,6 @@
     "### Functions:\n",
     "\n",
     "  * [empty](https://docs.scipy.org/doc/numpy/reference/generated/numpy.empty.html)\n",
-    "  * [meshgrid](https://docs.scipy.org/doc/numpy/reference/generated/numpy.meshgrid.html)\n",
```
Comment (kaedonkers) [inline course_content/notebooks/numpy_intro.ipynb]: Why was this removed?

File: course_content/notebooks/numpy_intro.ipynb (hunk 2 above)
Comment (dkillick) [reply course_content/notebooks/numpy_intro.ipynb]: If we do that we'll need to make sure **all** the exercises have numbers. As there are now _many_ exercises in the NumPy course (perhaps too many!) I'm not overly keen to add numbering. 

Also, we shouldn't let another PR hold up this PR. We can just rebase later PRs to make sure all the dust settles in the right place.

File: course_content/notebooks/numpy_intro.ipynb (hunk 3 above)
Comment (dkillick) [reply course_content/notebooks/numpy_intro.ipynb]: The courses are supposed to be useable both as taught training courses in a room and for a person to be able to work through on their own. This particular comment does not fit the second use-case!

But they can all still eat cake.

File: course_content/notebooks/numpy_intro.ipynb (hunk 5 above)
Comment (dkillick) [reply course_content/notebooks/numpy_intro.ipynb]: It's [now covered in the main course content](https://render.githubusercontent.com/view/ipynb?commit=5c2999f0a41c2e05f3c53ebe99c70fa1244e0dad&enc_url=68747470733a2f2f7261772e67697468756275736572636f6e74656e742e636f6d2f646b696c6c69636b2f636f75727365732f356332393939663061343163326530356633633533656265393963373066613132343465306461642f636f757273655f636f6e74656e742f6e6f7465626f6f6b732f6e756d70795f696e74726f2e6970796e62&nwo=dkillick%2Fcourses&path=course_content%2Fnotebooks%2Fnumpy_intro.ipynb&repository_id=39770343&repository_type=Repository#Generating-2D-coordinate-arrays). There's no point in also having it in further reading and it's sort of misleading to do so as the further reading section is supposed to contain links to content _not_ covered by the rest of the course!

File: course_content/notebooks/numpy_intro.ipynb (hunk 4 above)
Comment (dkillick) [reply course_content/notebooks/numpy_intro.ipynb]: This too is [now covered in the main course content](http://nbviewer.jupyter.org/github/SciTools/courses/blob/master/course_content/notebooks/numpy_intro.ipynb?create=1#Masked-Arrays).
